instrumentation/instalogrus/instalogrus.go

```diff
@@ -0,0 +1,71 @@
+// (c) Copyright IBM Corp. 2021
+// (c) Copyright Instana Inc. 2021
+
+package instalogrus
+
+import (
+	instana "github.com/instana/go-sensor"
+	"github.com/opentracing/opentracing-go"
+	"github.com/sirupsen/logrus"
+)
+
+type hook struct {
+	sensor *instana.Sensor
+}
+
+// NewHook returns a new logrus.Hook to instrument logger with Instana
+func NewHook(sensor *instana.Sensor) *hook {
+	return &hook{
+		sensor: sensor,
+	}
+}
+
+// Levels returns the list of log levels to be sent to Instana
+func (h *hook) Levels() []logrus.Level {
+	return []logrus.Level{logrus.ErrorLevel, logrus.WarnLevel}
+}
+
+// Fire forwards the logrus.Entry to Instana
+func (h *hook) Fire(entry *logrus.Entry) error {
+	if entry.Context == nil {
+		h.sensor.Logger().Debug("ignoring logrus.Entry without context.Context")
+		return nil
+	}
+
+	parent, ok := instana.SpanFromContext(entry.Context)
+	if !ok {
+		return nil
+	}
+
+	msg, err := entry.Bytes()
+	if err != nil {
+		h.sensor.Logger().Error("failed to obtain logrus.Entry data:", err)
+		return nil
+	}
+
+	h.sensor.Tracer().StartSpan("log.go",
+		opentracing.ChildOf(parent.Context()),
+		opentracing.StartTime(entry.Time),
+		opentracing.Tags{
+			"log.level":   convertLevel(entry.Level),
+			"log.message": string(msg),
+		},
+	).FinishWithOptions(opentracing.FinishOptions{
+		FinishTime: entry.Time,
+	})
+
+	return nil
+}
+
+func convertLevel(lvl logrus.Level) string {
```

Comment:
Not sure if this is better 😃  Just like an idea in order to not create extra mappings/constants, we can use something like :

```
func convertLevel(lvl logrus.Level) string {
	switch lvl {
	case logrus.ErrorLevel, logrus.WarnLevel, logrus.InfoLevel:
		return strings.ToUpper(lvl.String())
	default:
		return "DEBUG"
	}
}
```

Reply:
I'm usually trying to avoid relying on constant values, since this would introduce an implicit coupling between packages, i.e. we won't be able to change let's say `WARN` to `WARNING`. I see your point that it would be shorter, but still think that an explicit version is less error-prone and more explanatory.